src/serialize/json/denormalize.ts

```diff
@@ -0,0 +1,506 @@
+/*!
+This file is part of CycloneDX JavaScript Library.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+SPDX-License-Identifier: Apache-2.0
+Copyright (c) OWASP Foundation. All Rights Reserved.
+*/
+
+import { PackageURL } from 'packageurl-js'
+
+import { AttachmentEncoding, ComponentScope, ComponentType, ExternalReferenceType, HashAlgorithm } from '../../enums'
+import * as Models from '../../models'
+import type { Protocol, Protocol as Spec } from '../../spec'
+import { Format, SpecVersionDict, UnsupportedFormatError } from '../../spec'
+import type { JSONDenormalizerOptions, JSONDenormalizerWarning, PathType } from '../types'
+
+interface JSONDenormalizerContext {
```

Comment:
why is this needed? why are these three values encapsulated into a record? 
i found rare usage - and none was justified. all usage i found was artificially complex.